sharedse/src/commonMain/kotlin/com/ustadmobile/sharedse/network/containeruploader/UploadJobRunner.kt

```diff
@@ -0,0 +1,143 @@
+package com.ustadmobile.sharedse.network.containeruploader
+
+import com.ustadmobile.core.account.Endpoint
+import com.ustadmobile.core.db.JobStatus
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.networkmanager.defaultHttpClient
+import com.ustadmobile.core.util.UMFileUtil
+import com.ustadmobile.lib.db.entities.ConnectivityStatus
+import com.ustadmobile.lib.db.entities.ContainerUploadJob
+import com.ustadmobile.lib.db.entities.ContainerWithContainerEntryWithMd5
+import com.ustadmobile.sharedse.network.NetworkManagerBle
+import io.ktor.client.request.header
+import io.ktor.client.request.parameter
+import io.ktor.client.request.post
+import io.ktor.client.request.url
+import io.ktor.client.statement.HttpStatement
+import io.ktor.http.HttpStatusCode
+import io.ktor.network.sockets.SocketTimeoutException
+import io.ktor.utils.io.errors.IOException
+import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.kodein.di.DI
+import org.kodein.di.DIAware
+import org.kodein.di.instance
+import org.kodein.di.on
+
+class UploadJobRunner(private val containerUploadJob: ContainerUploadJob, private val endpointUrl: String, override val di: DI) : DIAware {
```

Comment:
If the upload job fails there needs to be a short delay before trying again (e.g. 1second). That should be a parameter to uploadjobrunner (a default can be defined as a constant). To make the tests run faster, you could reduce that delay.